### ISSUE TITLE
Tab navigation

### ISSUE THREAD

**Issue by MarufZak** [kamranahmedse/developer-roadmap#5354]
### What is the URL where the issue is happening

https://roadmap.sh/

### What browsers are you seeing the problem on?

Chrome

### Describe the Bug

This bug violates WCAG criteria 2.4.3 Focus Order (Level A)

As a keyboard user i noticed a small bug in the navigation component. When you reach a focus where active element is burger icon, the state is:

<img width="838" alt="Screenshot 2024-03-16 at 14 27 35" src="https://github.com/kamranahmedse/developer-roadmap/assets/72617743/0bb7b3ef-940b-4265-8f83-0e16bcfe7b26">

but when tabbing one more time, the visual indication of where the tab disappears:

<img width="838" alt="Screenshot 2024-03-16 at 14 29 20" src="https://github.com/kamranahmedse/developer-roadmap/assets/72617743/3ce8bdaa-1371-455d-ae9c-95d44f4e9864">

### Participation

- [X] I am willing to submit a pull request for this issue.
